### DISCUSSION TITLE
Order too complex..?

### DISCUSSION THREAD

**Discussion by Grelot11** [O365/python-o365#888] [Q&A] [answered]
hi,

im trying to do, what appear to me as simple, a query that filter on subject and order by datetime received..
i can do both separately but not together.. or did i missed the right way to get new emails first instead of olds one..?

`query = query.on_attribute('subject').contains('bla').order_by('receivedDateTime',ascending=False)`

this give me a "restriction or order is too complex" error..
thanks in advance..

**Comment by alejcas** [comment, marked as answer]
MS Graph docs are not very clear on what filters and orders can be done. You can try in ms graph explorer. If it works there it should work here.

In my experience contains filter can be very restrictive.

**Reply by Grelot11** [reply]
i'm using msoffice365protocol, sorry forgot to tell or do it make a difference..?

but i did check on ms graph and it appear at v1.0 you can't filter and orderby on the same query..
